### DISCUSSION TITLE
Example/sample for "Nordic Thingy:53" battery-operated prototype hardware?

### DISCUSSION THREAD

**Discussion by Hedda** [ffenix113/zigbee_home#6] [Ideas]
Please consider building POC firmware samples, for example a Zigbee Weather Station for the "Nordic Thingy:53" prototyping hardware:

https://www.nordicsemi.com/Nordic-news/2022/06/Nordic-Thingy53

https://www.nordicsemi.com/Products/Development-hardware/Nordic-Thingy-53

https://www.nordicsemi.com/-/media/Software-and-other-downloads/Product-Briefs/Nordic-Thingy53-PB-1.2.pdf

![image](https://github.com/ffenix113/zigbee_home/assets/6320001/cb54e312-59fa-4a16-89bc-df7340608406)

https://www.youtube.com/watch?v=PQhlOITwQTo

https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/ug_thingy53.html

Nordic Semiconductor themselves uses this Thingy53 dev kit hardware in their getting started documentation to help learn the platform:

https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/device_guides/working_with_nrf/nrf53/thingy53.html#ug-thingy53

https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/applications/zigbee_weather_station/README.html

https://academy.nordicsemi.com/flash-instructions-for-the-thingy53/

The Nordic Thingy:53™ is an easy-to-use IoT prototyping platform. It makes it possible to create prototypes and proofs-of-concept without building custom hardware. The Thingy:53 is built around the nRF5340 SoC. The processing power and memory size of its dual Arm Cortex-M33 processors enables it to run embedded machine learning (ML) models directly on the device.

Contents
*  Thingy:53
*  Protective Enclosure
* Current measurement and debug board (Does not include cables)
* Getting-started card (nordic.semi.com/start-thingy53)

Key features
* Supports Bluetooth LE, Bluetooth mesh, Thread, Zigbee, Matter, proprietary 2.4 GHz, and NFC
* Enabled for embedded machine learning (ML)
* Environmental sensor for temperature, humidity, air quality and air pressure
* Color and light sensor
* Low-power accelerometer and inertial measurement unit (IMU)
* High-quality MEMS microphone and buzzer
* User-programmable buttons and RGB LED
* Rechargeable Li-Po battery

The nRF Edge Impulse app enables users to connect their Nordic Thingy:53 to their Edge Impulse studio account through a mobile device. It allows them to wirelessly transfer sensor data over Bluetooth LE to the mobile device and upload it to the cloud for training and download trained ML models to the Thingy:53 for deployment and inferencing. The app also acts as the GUI for viewing inferencing results from a running ML model.

The Bluetooth Low Energy (LE) radio allows updating firmware and communication over Bluetooth LE, and the radio also supports other protocols like Bluetooth mesh, Thread, Zigbee, and proprietary 2.4 GHz protocols. The Thread protocol compatibility also makes it a great choice when developing products for the new Matter ecosystem. Matter is a standardized application layer for connected home applications, using the Internet Protocol (IP) as the network layer. 

The Thingy:53 also includes many different integrated sensors, like environmental-, and color and light sensors, accelerometers, and a magnetometer, which all can be taken advantage of without additional hardware. It is powered by a rechargeable Li-Po battery that can be charged via USB-C. There is also an external 4-pin JST connector compatible with the Stemma/Qwiic/Grove standards for hardware accessories.

Every Thingy:53 comes pre-installed with the firmware to work with the [nRF Edge Impulse](https://www.nordicsemi.com/Products/Development-tools/nRF-Edge-Impulse) app for iOS and Android. Also released with the Thingy:53 is the [nRF Programmer](https://www.nordicsemi.com/Products/Development-tools/nRF-Programmer) app for updating the device firmware directly over the air from a mobile device.

![image](https://github.com/ffenix113/zigbee_home/assets/6320001/735ad1d6-c684-4e50-9851-bb0bde03c482)

![image](https://github.com/ffenix113/zigbee_home/assets/6320001/8771ef5f-d164-45e9-b63c-a289d0559c5c)

![image](https://github.com/ffenix113/zigbee_home/assets/6320001/cc3be466-248f-483c-a2db-227e5e314941)

PS: NordicSemi also has the nRF5340 DK Development Kit as well but that is much more intimidating so can put some users off:

https://www.nordicsemi.com/Products/Development-hardware/nRF5340-DK

**Comment by Hedda** [comment]
Just me or does this official "Nordic Thingy:53" (based on [nRF5340](https://github.com/ffenix113/zigbee_home/discussions/20)) sound like it could be a great reference design device for this project?

https://www.hackster.io/news/nordic-thingy-53-provides-rapid-prototyping-for-embedded-machine-learning-and-iot-d141a6f96099

https://www.hackster.io/mahmood-ul-hassan/how-to-interface-nordic-thingy-53-with-pca9685-6c6291

https://community.element14.com/products/roadtest/b/blog/posts/nordic-thingy-53-first-impression

https://community.element14.com/products/roadtest/rt/roadtests/608/nordic_thingy53_with#pifragment-18971=4&pifragment-18972=7

https://www.youtube.com/watch?v=xvfM92dgpJI

https://www.youtube.com/watch?v=Jbrwv1zgRoI

https://edge-impulse.gitbook.io/docs/edge-ai-hardware/mcu/nordic-semi-thingy53

Tthe following set of sensors comes included built-int the Nordic Thingy53 out-of-the-box:

* Accelerometer
* Microphone
* Magnetometer
* Light
* Temperature, Pressure, Humidity and Air quality
* Inertial (Accelerometer and Magnetometer)
* Light and Environment
* Light and Inertial
* Environment and Inertial

**Reply by ffenix113** [reply]
Yes, it would be a great example. Once I will have funds available to purchase Thingy:53 and nrf5340 DK - I will get right to it!

**Reply by ffenix113** [reply]
I've bought the 5340 dev board from Raytac, and seems that the nRF52840 DK can debug 5340. Now I will be able to work on the support for 5340.

Unfortunately I am rather busy with things outside of this project, so I can't spend as much time on this as I would've liked(at least for next couple of weeks).
